File: data/src/main/java/com/roman/kubik/spivanyklicejista/data/database/AppDatabase.java
package com.roman.kubik.spivanyklicejista.data.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;

import com.roman.kubik.spivanyklicejista.data.category.CategoryDao;
import com.roman.kubik.spivanyklicejista.data.category.CategoryEntity;
import com.roman.kubik.spivanyklicejista.data.chord.ChordDao;
import com.roman.kubik.spivanyklicejista.data.chord.ChordEntity;
import com.roman.kubik.spivanyklicejista.data.favourite.FavouriteDao;
import com.roman.kubik.spivanyklicejista.data.favourite.FavouriteEntity;
import com.roman.kubik.spivanyklicejista.data.song.SongDao;
import com.roman.kubik.spivanyklicejista.data.song.SongEntity;

/**
 * Room database representation
 * Created by kubik on 1/14/18.
 */
@Database(entities = {SongEntity.class, CategoryEntity.class, ChordEntity.class, FavouriteEntity.class}, version = 2)
public abstract class AppDatabase extends RoomDatabase {

    public abstract SongDao songDao();

    public abstract CategoryDao categoryDao();

    public abstract ChordDao chordDao();

    public abstract FavouriteDao favouriteDao();

}

File: app/src/main/java/com/roman/kubik/songer/data/local/database/AppDatabase.kt
package com.roman.kubik.songer.data.local.database

import androidx.room.Database
import androidx.room.RoomDatabase
import com.roman.kubik.songer.data.local.category.CategoryEntity
import com.roman.kubik.songer.data.local.favourite.FavouriteEntity
import com.roman.kubik.songer.data.local.history.HistoryEntity
import com.roman.kubik.songer.data.local.song.SongDao
import com.roman.kubik.songer.data.local.song.SongEntity

/**
 * Room database representation
 */
@Database(entities = [SongEntity::class, CategoryEntity::class, FavouriteEntity::class, HistoryEntity::class], version = 5, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun songDao(): SongDao
//    abstract fun categoryDao(): CategoryDao?
//    abstract fun favouriteDao(): FavouriteDao?
//    abstract fun historyDao(): HistoryDao?
}